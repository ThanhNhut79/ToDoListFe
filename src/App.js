import React, { useState, useEffect } from "react"
import Header from "./Header"
import TodoList from "./TodoList"
// import { AdapterDateFns } from "@mui/x-date-pickers/AdapterDateFns"
// import { LocalizationProvider } from "@mui/x-date-pickers/LocalizationProvider"
import { LocalizationProvider } from "@mui/lab"
import AdapterDateFns from "@mui/lab/AdapterDateFns"
import axios from "axios"
import moment from "moment"

const App = () => {
  const [todoList, setTodoList] = useState([])
  const [value, setValue] = useState(null)
  const [userInput, setUserInput] = useState("")

  useEffect(() => {
    const fetchTodo = async () => {
      try {
        await axios
          .get("http://localhost:7000")
          .then((response) => setTodoList(response.data))
      } catch (error) {
        console.log(error)
      }
    }
    fetchTodo()
  }, [])

  const handleFilter = async () => {
    try {
      await axios
        .delete("http://localhost:7000")
        .then((response) => response.data)
        .then((data) => setTodoList(data))
    } catch (error) {
      console.log(error)
    }
  }

  const addTask = async (userInput, value) => {
    if (!value) return
    const date = moment(value).format("MM/DD/YYYY")
    try {
      await axios
        .post("http://localhost:7000", { task: userInput, date })
        .then((response) => response.data)
        .then((data) => {
          let copy = [...todoList]
          setTodoList([...copy, data.newTodo])
          setUserInput("")
          setValue(null)
        })
    } catch (error) {
      console.log(error)
    }
  }

  if (!todoList || todoList.length === 0) {
    return <p>Loading...</p>
  }

  return (
    <LocalizationProvider dateAdapter={AdapterDateFns}>
      <div className='App'>
        <Header />
        <TodoList
          todoList={todoList}
          setTodoList={setTodoList}
          handleFilter={handleFilter}
          addTask={addTask}
          userInput={userInput}
          setUserInput={setUserInput}
          value={value}
          setValue={setValue}
        />
      </div>
    </LocalizationProvider>
  )
}

export default App
