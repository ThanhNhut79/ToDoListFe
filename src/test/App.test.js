import axios from "axios"
import renderer, { act } from "react-test-renderer"
import App from "../App"

const todoList = [
  {
    completed: false,
    task: "Give dog a bath",
    _id: "62986866968e18bbf3565d2e",
    date: "06/23/2022",
  },
  {
    completed: false,
    task: "Give dog a bath",
    _id: "62996bee1da17815189dabe0",
    date: "06/23/2022",
  },
]

jest.mock("axios")

describe("App component", () => {
  test("App test works", async () => {
    let tree
    axios.get.mockResolvedValue({ data: todoList })
    await act(async () => {
      tree = renderer.create(<App />)
    })
    expect(tree.toJSON()).toMatchSnapshot()
  })
})
