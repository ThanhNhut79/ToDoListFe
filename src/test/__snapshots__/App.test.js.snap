// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component App test works 1`] = `
<div
  className="App"
>
  <h2
    className="MuiTypography-root MuiTypography-h2 css-1sra7t5-MuiTypography-root"
  >
    To Do List
  </h2>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation6 paper css-azc51k-MuiPaper-root"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4 css-480o17-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-md-4 css-14a3zbu-MuiGrid-root"
      >
        <button
          className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root todo-btn css-1e6y48t-MuiButtonBase-root-MuiButton-root"
          data-test="todo-btn"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <p
            className="MuiTypography-root MuiTypography-body1  todoTask css-ahj2mt-MuiTypography-root"
            data-test="todoTask"
          >
            Give dog a bath
          </p>
          <span
            className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <p
          className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          End day: 
          06/23/2022
        </p>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-md-4 css-14a3zbu-MuiGrid-root"
      >
        <button
          className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root todo-btn css-1e6y48t-MuiButtonBase-root-MuiButton-root"
          data-test="todo-btn"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <p
            className="MuiTypography-root MuiTypography-body1  todoTask css-ahj2mt-MuiTypography-root"
            data-test="todoTask"
          >
            Give dog a bath
          </p>
          <span
            className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <p
          className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          End day: 
          06/23/2022
        </p>
      </div>
    </div>
    <div
      className="css-nen11g-MuiStack-root"
    >
      <form
        className="my-form-control"
        onSubmit={[Function]}
      >
        <div
          className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          data-test="task"
        >
          <label
            className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink={false}
            htmlFor=":r0:"
            id=":r0:-label"
          >
            Enter task
          </label>
          <div
            className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              disabled={false}
              id=":r0:"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="Enter task..."
              required={false}
              type="text"
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-1ftyaf0"
              >
                <span>
                  Enter task
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          data-test="date"
        >
          <label
            className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink={false}
            htmlFor=":r1:"
            id=":r1:-label"
          >
            Date
          </label>
          <div
            className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              aria-label="Choose date"
              aria-readonly={true}
              autoFocus={false}
              className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              id=":r1:"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              readOnly={true}
              required={false}
              type="text"
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-1ftyaf0"
              >
                <span>
                  Date
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <button
          className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButtonBase-root submit-btn css-jh47zj-MuiButtonBase-root-MuiButton-root"
          disabled={false}
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="submit"
        >
          Submit
          <span
            className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </form>
      <button
        className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
        data-test="delete-btn"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "marginTop": "10px",
          }
        }
        tabIndex={0}
        type="button"
      >
        Clear Complete
        <span
          className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
